packages/search-ui/src/DebounceManager.js

```diff
@@ -1,30 +1,58 @@
 import debounceFn from "debounce-fn";
 
-export default class DebounceManager {
+class DebounceManager {
   debounceCache = {};
 
-  /*
-  The purpose of this is to:
-  Dynamically debounce and cache a debounced version of a function at the time of calling that function. This avoids
-  managing debounced version of functions locally.
-
-  Assumption:
-  Functions are debounced on a combination of unique function and wait times. So debouncing won't work on
-  subsequent calls with different wait times or different functions. That also means that the debounce manager
-  can be used for different functions in parallel, and keep the two functions debounced separately.
-
-  */
-  runWithDebounce(wait, fn, ...parameters) {
+  /**
+   * Dynamically debounce and cache a debounced version of a function at the time of calling that function. This avoids
+   * managing debounced version of functions locally.
+   *
+   * In other words, debounce usually works by debouncing based on
+   * referential identity of a function. This works by comparing provided function names.
+   *
+   * This also has the ability to short-circuit a debounce all-together, if no wait
+   * time is provided.
+   *
+   * Assumption:
+   * Functions are debounced on a combination of unique function name and wait times. So debouncing won't work on
+   * subsequent calls with different wait times or different functions. That also means that the debounce manager
+   * can be used for different functions in parallel, and keep the two functions debounced separately.
+   *
+   *
+   * @param {number} wait Milliseconds to debounce. Executes immediately if falsey.
+   * @param {function} fn Function to debounce
+   * @param {function} functionName Name of function to debounce, used to create a unique key
+   * @param  {...any} parameters Parameters to pass to function
+   */
+  runWithDebounce(wait, functionName, fn, ...parameters) {
     if (!wait) {
       return fn(...parameters);
     }
 
-    const key = fn.toString() + wait.toString();
+    const key = functionName + wait.toString();
     let debounced = this.debounceCache[key];
     if (!debounced) {
       this.debounceCache[key] = debounceFn(fn, { wait });
       debounced = this.debounceCache[key];
     }
     debounced(...parameters);
   }
+
+  cancelByName(functionName) {
```

Comment:
I'd love to see this method get a docblock as well, however short/small!

Reply:
Ah, yes. Good call, thanks for keeping me on my toes!